### PR TITLE
Add a plugin that visualizes the maximum horizontal compressive stress.

### PR DESCRIPTION
**Pull request by bangerth**
This only took ~6 weeks longer than I had promised to @dsarahstamps . But I think the pictures are good.

I took the time to really describe in detail what the plugin actually shows, given that I had trouble finding a reasonable description I could understand in the literature. This may well be the best documented postprocessor plugin now.

### CODE REVIEW

**Comment by bangerth** [comment]
@dsarahstamps -- here is my final version of the visualization.
horizontal-stress

**Comment by bangerth** [comment]
(The picture is for a double convection roll with material rusing up in the middle. The other pictures are in the pull request and will be part of the manual.)

**Comment by gassmoeller** [comment]
This does not compile on the tester. Is this a problem that would be solved by using deal.II 8.4 instead, or is it something more serious?

**Comment by bangerth** [comment]
I'll have to take care of this when back in early June. For my own record, this is the error:
```
/aspect-tester/aspect/source/postprocess/visualization/maximum_horizontal_compressive_stress.cc: In instantiation of â€˜void aspect::Postprocess::VisualizationPostprocessors::MaximumHorizontalCompressiveStress<dim>::compute_derived_quantities_vector(const std::vector<dealii::Vector<double> >&, const std::vector<std::vector<dealii::Tensor<1, dim> > >&, const std::vector<std::vector<dealii::Tensor<2, dim, double> > >&, const std::vector<dealii::Point<dim, double> >&, const std::vector<dealii::Point<dim, double> >&, std::vector<dealii::Vector<double> >&) const [with int dim = 2]â€™:
/aspect-tester/aspect/source/postprocess/visualization/maximum_horizontal_compressive_stress.cc:243:7:   required from here
/aspect-tester/aspect/source/postprocess/visualization/maximum_horizontal_compressive_stress.cc:122:85: error: no match for â€˜operator*â€™ (operand types are â€˜std::array<dealii::Tensor<1, 2, double>, 1ul>::value_type {aka dealii::Tensor<1, 2, double>}â€™ and â€˜dealii::SymmetricTensor<2, 2>â€™)
                                                           (orthogonal_directions[0] *
                                                                                     ^
```
I suspect that I'm calling functions that were added post-8.4.

**Comment by bangerth** [comment]
I fixed the compilation with older deal.II 8.4. Ready for re-review.

File: doc/modules/changes.h [hunk 1]
```diff
@@ -36,6 +37,11 @@
  * <br>
  * (Juliane Dannberg, 2016/05/19)
  *
+ * <li> New: There is now a visualization plugin to visualize the maximum
+ * horizontal component of the compressive strength.
```
Comment (gassmoeller) [inline doc/modules/changes.h]: Do you mean compressive stress? I usually associate 'strength' with a plastic failure criterion.

File: source/postprocess/visualization/maximum_horizontal_compressive_stress.cc [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/postprocess/visualization/maximum_horizontal_compressive_stress.h>
+#include <aspect/simulator_access.h>
```
Comment (gassmoeller) [inline source/postprocess/visualization/maximum_horizontal_compressive_stress.cc]: simulator_access.h is already included in the header file. I guess this is a copy-paste error and we have another postprocessor that has the same double include?

File: source/postprocess/visualization/maximum_horizontal_compressive_stress.cc [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+
+            const double eta = out.viscosities[q];
+
+            // first compute the stress tensor, ignoring the pressure
+            // for the moment (the pressure has no effect on the
+            // direction since it just adds a multiple of the identity
+            // matrix to the stress, but because it is large, it may
+            // lead to numerical instabilities)
+            //
+            // note that the *compressive* strength is simply the
```
Comment (gassmoeller) [inline source/postprocess/visualization/maximum_horizontal_compressive_stress.cc]: compressive stress?

**Comment by gassmoeller** [comment]
Apart from the minor wording issues and the one double include this is ready to go. Feel free to improve the marked places and merge yourself.

**Pull request merged**